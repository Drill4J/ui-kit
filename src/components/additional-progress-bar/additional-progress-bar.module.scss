@import '../../style-guide/index';

.additional-progress-bar {
  height: 12px;
  border-radius: 4px;
  background: map-get($monochrome, light-tint);

  &__progress {
    height: 12px;
    border-radius: 4px;
    background: map-get($data-visualization, build-cover);

    &--type {
      border-radius: 0 4px 4px 0;

      &:hover {
        transform: scaleY(1.7);
      }

      &--primary {
        background: map-get($data-visualization, scope-cover);
      }

      &--secondary {
        background: map-get($data-visualization, overlapping);
      }
    }
  }
}
