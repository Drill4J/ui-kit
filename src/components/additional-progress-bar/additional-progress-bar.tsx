import styled, { css } from 'styled-components';

import { COLORS } from '../../colors';

interface Props {
  className?: string;
  value: string;
  type?: 'primary' | 'secondary';
  testContext?: string;
}

export const AdditionalProgressBar = ({
  className,
  value,
  type,
  testContext,
}: Props) => (
  <Wrapper className={className} data-test={`additional-progress-bar:${testContext || type}`}>
    <Progress style={{ width: value }} type={type} />
  </Wrapper>
);

const Wrapper = styled.div`
  height: 12px;
  border-radius: 4px;
  background: ${COLORS.MONOCHROME.LIGHT_TINT};
`;

const Progress = styled.div<{ type?: 'primary' | 'secondary' }>`
  height: 12px;
  border-radius: 4px;
  background: ${COLORS.DATA_VISUALIZATION.BUILD_COVER};
  ${({ type }) => [
    typeof type !== 'undefined' && css`
      border-radius: 0 4px 4px 0;

      &:hover {
        transform: scaleY(1.7);
      }
    `,
    type === 'primary' && css`background: ${COLORS.DATA_VISUALIZATION.SCOPE_COVER}`,
    type === 'secondary' && css`background: ${COLORS.DATA_VISUALIZATION.OVERLAPPING}`]}
`;
