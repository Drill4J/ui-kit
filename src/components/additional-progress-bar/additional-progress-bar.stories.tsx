import React from 'react';
import { Meta, Story } from '@storybook/react';
import 'twin.macro';

import { AdditionalProgressBar } from './additional-progress-bar';

export default {
  title: 'AdditionalProgressBar',
  component: AdditionalProgressBar,
  argTypes: {
    value: {
      control: {
        type: 'range',
        min: 0,
        max: 300,
        step: 1,
      },
    },
    type: {
      control: {
        type: 'radio',
      },
      options: ['primary', 'secondary'],
    },
  },
} as Meta;

const TemplateAdditionalProgressBar: Story = (args) => <AdditionalProgressBar value="100" {...args} />;

export const DefaultAdditionalProgressBar: Story = TemplateAdditionalProgressBar.bind({});

export const PrimaryAdditionalProgressBar: Story = TemplateAdditionalProgressBar.bind({});
PrimaryAdditionalProgressBar.args = { type: 'primary' };

export const SecondaryAdditionalProgressBar: Story = TemplateAdditionalProgressBar.bind({});
SecondaryAdditionalProgressBar.args = { type: 'secondary' };
