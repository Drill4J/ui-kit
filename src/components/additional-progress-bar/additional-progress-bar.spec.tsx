import React from 'react';
import renderer from 'react-test-renderer';
import { AdditionalProgressBar } from './additional-progress-bar';

describe('AdditionalProgressBar', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<AdditionalProgressBar value="50%" type="primary" testContext="snapshot" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
