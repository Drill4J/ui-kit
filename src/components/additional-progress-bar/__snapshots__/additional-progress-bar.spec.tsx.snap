// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondaryProgressBar should match snapshot 1`] = `
<div
  className="sc-bdvvaa YEirn"
  data-test="additional-progress-bar:undefined"
>
  <div
    className="sc-gsDJrp jOZeBx"
    style={
      Object {
        "width": "50px",
      }
    }
  />
</div>
`;
